packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2098,6 +2167,32 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
   @protected
   bool get alwaysNeedsCompositing => false;
 
+  late bool _wasRepaintBoundary;
+
+  /// Update the composited layer owned by this render object.
+  ///
+  /// This method is called by the framework when [isRepaintBoundary] is true.
+  ///
+  /// If [oldLayer] is `null`, this method must return a new [OffsetLayer]
```

Comment:
Document that the framework will dispose this layer if its no longer needed to discorage people from holding a reference to that that may become invalid.

Reply:
done